src/FuelClient.cc

```diff
@@ -410,10 +415,46 @@ Result FuelClient::UploadModel(const std::string &_pathToModelDir,
   // \todo(nkoenig) The ign-fuelserver expects an integer number for the
   // license information. The fuelserver should be modified to accept
   // a string. Otherwise, we have to bake into each client a mapping of
-  // license name to integer. For now, we are making a model
-  // "Creative Commons - Public Domain
-  // if (meta.has_legal()) {....}
-  form.emplace("license", "1");
+  // license name to integer.
+  //
+  // If there is no license information, then default to
+  // "Creative Commons - Public Domain"
```

Comment:
I think a downstream user could easily forget to call the `PopulateLicenses` function and end up with public domain by mistake.

My suggestion would be to make `PopulateLicenses` a private function and call it once when the user tries to upload models with custom licenses. And fail to upload if they explicitly asked for a license but it wasn't able to get licenses from the server.

Something like this:

```.cpp
if (meta.has_legal())
{
  if (this->dataPtr->licenses.empty())
  {
    this->PopulateLicenses();
    if (this->dataPtr->licenses.empty())
    {
      // Fail
    }
  }
  // Use license from legal
}
else
{
  // Default to public domain
}
```

Reply:
Good idea. Updated here: aba1a14d3862b0991fd27d74ef049acafbd8c2db